src/gc.c

```diff
@@ -3040,6 +3231,150 @@ JL_DLLEXPORT jl_value_t *jl_gc_alloc_3w(void)
     return jl_gc_alloc(ptls, sizeof(void*) * 3, NULL);
 }
 
+JL_DLLEXPORT int jl_gc_enable_conservative_gc_support()
+{
+    static jl_mutex_t conservative_gc_lock;
+    static_assert(jl_buff_tag % GC_PAGE_SZ == 0,
+        "jl_buff_tag must be a multiple of GC_PAGE_SZ");
+    if (jl_is_initialized()) {
+        JL_LOCK_NOGC(&conservative_gc_lock);
+        int result = support_conservative_marking;
+        support_conservative_marking = 1;
+        JL_UNLOCK_NOGC(&conservative_gc_lock);
+        if (!result) {
+            // Do a full collection to ensure that age bits are updated
+            // properly. We don't have to worry about race conditions
+            // for this part, as allocation itself is unproblematic and
+            // a collection will wait for safepoints.
+            jl_gc_collect(1);
+        }
+        return result;
+    } else {
+        int result = support_conservative_marking;
+        support_conservative_marking = 1;
+        return result;
+    }
+}
+
+JL_DLLEXPORT int jl_gc_conservative_gc_support_enabled()
+{
+    return support_conservative_marking;
```

Comment:
Since the other one has lock/atomic, this one should be as well mainly for consistency. `jl_atomic_load_acquire`.

Reply:
Will do. It's not strictly necessary, as this is a benign data race, but I agree that it would be less confusing.